hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/execution/EvmTxProcessor.java

```diff
@@ -190,7 +194,10 @@ protected TransactionProcessingResult execute(
             initialFrame.getSelfDestructs().forEach(updater::deleteAccount);
 
             if (isSideCarTypeEnabled(SidecarType.CONTRACT_STATE_CHANGE)) {
-                stateChanges = ((HederaWorldState.Updater) updater).getFinalStateChanges();
+                stateChanges =
+                        sidecarInstrumentation.captureDurationSplit(
+                                SidecarType.CONTRACT_STATE_CHANGE,
+                                () -> ((HederaWorldState.Updater) updater).getFinalStateChanges());
```

Comment:
When it comes to capturing state changes, we should also maybe add tracking in [HederaOperationUtil#cacheExistingValue](https://github.com/hashgraph/hedera-services/blob/8d9d4c3dc5f6ec93f75444ecad5bffcf5aac2971/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/operation/HederaOperationUtil.java#L123-L137) --- that method gets called each time we execute an [SStore](https://github.com/hashgraph/hedera-services/blob/8d9d4c3dc5f6ec93f75444ecad5bffcf5aac2971/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/operation/HederaSStoreOperation.java#L78-L80) / [SLoad](https://github.com/hashgraph/hedera-services/blob/8d9d4c3dc5f6ec93f75444ecad5bffcf5aac2971/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/operation/HederaSLoadOperation.java#L66-L67) operation, and it basically adds the initially read value of the particular contract slot in the state changes map, which we are getting here.

Reply:
That's a good idea as that cache is only used if the state change sidecar is enabled.  _Unfortunately_ the method is in a utility class (a class containing only static methods that is never instantiated) so the instrumentation can't be held by the class, plus the methods that _call_ this are in a complex area that I haven't looked at so I'm not comfortable going ahead to plumb it all though that.  Finally, I'm not sufficiently savvy to know if there's any kind of "sufficiently per-transaction global" - the sidecar instrumentation is created per-transaction - where I could hold the thing.  So ... let's leave this pending for future improvement.